### ISSUE TITLE
Full database syncup even when there is zero change. 

### ISSUE THREAD

**Issue by bipulkarnani** [orientechnologies/orientdb#8210]
### OrientDB Version: 2.2.33
### Java Version: 1.8
### OS: centOS

## Expected behavior  
We had three servers with the following `distributed-db-config.json`: 
```{                                                                                                                                                                                    
  "autoDeploy": true,                                                                                                                                                                
  "hotAlignment": true,                                                                                                                                                              
  "executionMode": "undefined",                                                                                                                                                      
  "readQuorum": 1,                                                                                                                                                                   
  "writeQuorum": "majority",                                                                                                                                                         
  "newNodeStrategy": "dynamic",                                                                                                                                                      
  "failureAvailableNodesLessQuorum": false,                                                                                                                                          
  "readYourWrites": false,                                                                                                                                                           
  "servers": {                                                                                                                                                                       
    "*": "master"                                                                                                                                                                    
  },                                                                                                                                                                                 
  "clusters": {                                                                                                                                                                      
    "internal": {                                                                                                                                                                    
    },                                                                                                                                                                               
    "index": {                                                                                                                                                                       
    },                                                                                                                                                                               
    "*": {                                                                                                                                                                           
      "servers": ["<NEW_NODE>"]                                                                                                                                                      
    }                                                                                                                                                                                
  }                                                                                                                                                                                  
}
```  
Our three servers are named: `odb1`, `odb2`, `odb3`.  I stopped `odb1` and then again started `odb1`. After starting `odb1`, `odb2` and `odb3` started syncing the entire database where `odb1` should do with one of `odb2` or `odb3`. I am attaching the entire log: 
```
                                                                                                                       
2018-04-12 13:03:10:982 INFO  [odb3] Reassignment of clusters for database 'paysense' completed (classes=13)           
2018-04-12 13:08:14:209 INFO  [10.10.15.141]:5701 [paysense-graph] [3.8.4] Accepting socket connection from /10.10.15.1
2018-04-12 13:08:14:210 INFO  [10.10.15.141]:5701 [paysense-graph] [3.8.4] Established socket connection between /10.10
2018-04-12 13:08:21:210 INFO  [10.10.15.141]:5701 [paysense-graph] [3.8.4]                                             
                                                                                                                       
Members [3] {                                                                                                          
        Member [10.10.15.156]:5701 - 36d7e021-305b-435c-b175-e274f417cf80                                              
        Member [10.10.15.141]:5701 - 1e700307-5d4f-42c2-80be-1d83c3c8f12c this                                         
        Member [10.10.15.179]:5701 - aa891e73-49ad-4563-807d-b1bcc8e6d17f                                              
}                                                                                                                      
 [ClusterService]                                                                                                      
2018-04-12 13:08:21:211 INFO  [odb3] Added new node id=Member [10.10.15.179]:5701 - aa891e73-49ad-4563-807d-b1bcc8e6d17
2018-04-12 13:08:23:303 INFO  [odb3]<-[ext:aa891e73-49ad-4563-807d-b1bcc8e6d17f] Receivedupdated about registered nodes
2018-04-12 13:08:23:460 INFO  [odb3]<-[odb1] Added node configuration id=Member [10.10.15.179]:5701 - aa891e73-49ad-456
2018-04-12 13:08:23:461 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                       
                                                                                                                       
+-------+--------+------------------------+-----+---------+-----------------+-----------------+-----------------------+
|Name   |Status  |Databases               |Conns|StartedOn|Binary           |HTTP    |UsedMemory             |         
+-------+--------+------------------------+-----+---------+-----------------+-----------------+-----------------------+
|odb3(*)|ONLINE  |paysense=ONLINE (MASTER)|3    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|149.83MB/1.99GB (7.37%)|
|odb2(@)|ONLINE  |paysense=ONLINE (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%) |
|odb1   |STARTING|                        |0    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|184.26MB/1.92GB (9.39%)|
+-------+--------+------------------------+-----+---------+-----------------+-----------------+-----------------------+
                                                                                                                       
2018-04-12 13:08:23:462 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                       
                                                                                                                       
+-------+------+------------------------+-----+---------+-----------------+-----------------+-----------------------+  
|Name   |Status|Databases               |Conns|StartedOn|Binary           |HTTP  |UsedMemory             |             
+-------+------+------------------------+-----+---------+-----------------+-----------------+-----------------------+  
|odb3(*)|ONLINE|paysense=ONLINE (MASTER)|3    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|149.83MB/1.99GB (7.37%)|  
|odb2(@)|ONLINE|paysense=ONLINE (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%) |  
|odb1   |      |                        |     |         |                 |  |                       |                 
+-------+------+------------------------+-----+---------+-----------------+-----------------+-----------------------+  
                                                                                                                       
2018-04-12 13:08:23:463 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                       
                                                                                                                       
+-------+--------+------------------------+-----+---------+-----------------+-----------------+-----------------------+
|Name   |Status  |Databases               |Conns|StartedOn|Binary           |HTTP    |UsedMemory             |         
+-------+--------+------------------------+-----+---------+-----------------+-----------------+-----------------------+
|odb3(*)|ONLINE  |paysense=ONLINE (MASTER)|3    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|149.83MB/1.99GB (7.37%)|
|odb2(@)|ONLINE  |paysense=ONLINE (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%) |
|odb1   |STARTING|                        |0    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|184.26MB/1.92GB (9.39%)|
+-------+--------+------------------------+-----+---------+-----------------+-----------------+-----------------------+
                                                                                                                       
2018-04-12 13:08:23:465 INFO  [odb3] Setting new distributed configuration for database: paysense (version=21)         
                                                                                                                       
CLUSTER CONFIGURATION [wQuorum: true] (LEGEND: X = Owner, o = Copy)

+--------+----+-----------+----------+-------+-------+-------------+                                                                                                        [939/992]
|        |    |           |          |MASTER |MASTER |   MASTER    |                                                                                                                 
|        |    |           |          |ONLINE |ONLINE |NOT_AVAILABLE|                                                                                                                 
|        |    |           |          |dynamic|dynamic|   dynamic   |                                                                                                                 
+--------+----+-----------+----------+-------+-------+-------------+                                                                                                                 
|CLUSTER |  id|writeQuorum|readQuorum| odb2  | odb3  |    odb1     |                                                                                                                 
+--------+----+-----------+----------+-------+-------+-------------+                                                                                                                 
|*       |    |     2     |    1     |   X   |   o   |      o      |                                                                                                                 
|device_2|  19|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|device_3|  20|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|e       |  13|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|e_1     |  14|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|index   |   1|     2     |    1     |       |       |             |                                                                                                                 
|internal|   0|     2     |    1     |       |       |             |                                                                                                                 
|knows   |  21|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|knows_3 |  24|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|person  |  25|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|person_1|  26|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|v       |   9|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
|v_3     |  12|     2     |    1     |   o   |   X   |      o      |                                                                                                                 
+--------+----+-----------+----------+-------+-------+-------------+                                                                                                                 
                                                                                                                                                                                     
                                                                                                                                                                                     
2018-04-12 13:08:23:539 INFO  [odb3]<-[odb2] Received updated status odb1.paysense=SYNCHRONIZING                                                                                     
2018-04-12 13:08:23:586 INFO  [odb3]<-[odb1] Received updated status odb1.paysense=ONLINE                                                                                            
2018-04-12 13:08:23:589 INFO  [odb3]<-[odb1] Received updated status odb3.paysense=NOT_AVAILABLE                                                                                     
2018-04-12 13:08:23:591 INFO  [odb3]<-[odb1] Received updated status odb2.paysense=NOT_AVAILABLE                                                                                     
2018-04-12 13:08:23:592 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                                                                                     
                                                                                                                                                                                     
+-------+--------+-------------------------------+-----+---------+-----------------+-----------------+-----------------------+                                                       
|Name   |Status  |Databases                      |Conns|StartedOn|Binary           |HTTP           |UsedMemory             |                                                         
+-------+--------+-------------------------------+-----+---------+-----------------+-----------------+-----------------------+                                                       
|odb3(*)|ONLINE  |paysense=NOT_AVAILABLE (MASTER)|3    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|149.83MB/1.99GB (7.37%)|                                                       
|odb2(@)|ONLINE  |paysense=ONLINE (MASTER)       |2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%) |                                                       
|odb1   |STARTING|                               |0    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|184.26MB/1.92GB (9.39%)|                                                       
+-------+--------+-------------------------------+-----+---------+-----------------+-----------------+-----------------------+                                                       
                                                                                                                                                                                     
2018-04-12 13:08:23:600 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                                                                                     
                                                                                                                                                                                     
+-------+--------+-------------------------------+-----+---------+-----------------+-----------------+-----------------------+                                                       
|Name   |Status  |Databases                      |Conns|StartedOn|Binary           |HTTP           |UsedMemory             |                                                         
+-------+--------+-------------------------------+-----+---------+-----------------+-----------------+-----------------------+                                                       
|odb3(*)|ONLINE  |paysense=NOT_AVAILABLE (MASTER)|3    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|149.83MB/1.99GB (7.37%)|                                                       
|odb2(@)|ONLINE  |paysense=NOT_AVAILABLE (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%) |                                                       
|odb1   |STARTING|                               |0    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|184.26MB/1.92GB (9.39%)|                                                       
+-------+--------+-------------------------------+-----+---------+-----------------+-----------------+-----------------------+                                                       
                                                                                                                                                                                     
2018-04-12 13:08:24:580 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):

+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|Name   |Status|Databases                      |Conns|StartedOn|Binary           |HTTP         |UsedMemory              |                                                            
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|odb3(*)|ONLINE|paysense=NOT_AVAILABLE (MASTER)|3    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|149.83MB/1.99GB (7.37%) |                                                        
|odb2(@)|ONLINE|paysense=NOT_AVAILABLE (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%)  |                                                        
|odb1   |ONLINE|paysense=ONLINE (MASTER)       |6    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|226.28MB/1.92GB (11.53%)|                                                        
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
                                                                                                                                                                                     
2018-04-12 13:08:24:587 INFO  [odb3] Trying to recover current server for database 'paysense'...                                                                                     
2018-04-12 13:08:24:589 INFO  [odb3] Current node is a MASTER for database 'paysense'                                                                                                
2018-04-12 13:08:24:590 WARNI [odb3]->[[odb1]] requesting delta database sync for 'paysense' on local server...                                                                      
2018-04-12 13:08:24:592 INFO  Found 0 records changed in last 0 operations                                                                                                           
2018-04-12 13:08:24:593 INFO  [odb3]->[odb1] Requesting database delta sync for 'paysense' LSN=LSN{segment=20, position=52}...                                                       
2018-04-12 13:08:24:613 INFO  [odb3]<-[odb1] Received updated status odb3.paysense=SYNCHRONIZING                                                                                     
2018-04-12 13:08:24:643 WARNI [odb3]<-[odb1] Error on installing database delta for 'paysense' (err=Requested database delta sync with LSN=LSN{segment=20, position=52} but not found
 in database)                                                                                                                                                                        
2018-04-12 13:08:24:643 INFO  [odb3] Requesting full sync for database 'paysense'...                                                                                                 
2018-04-12 13:08:24:643 INFO  [odb3]->[[odb1]] Requesting deploy of database 'paysense' on local server...                                                                           
2018-04-12 13:08:24:650 INFO  [odb3] Received updated status odb3.paysense=ONLINE                                                                                                    
2018-04-12 13:08:24:651 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                                                                                     
                                                                                                                                                                                     
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|Name   |Status|Databases                      |Conns|StartedOn|Binary           |HTTP         |UsedMemory              |                                                            
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|odb3(*)|ONLINE|paysense=SYNCHRONIZING (MASTER)|5    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|169.26MB/1.99GB (8.32%) |                                                        
|odb2(@)|ONLINE|paysense=NOT_AVAILABLE (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%)  |                                                        
|odb1   |ONLINE|paysense=ONLINE (MASTER)       |6    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|226.28MB/1.92GB (11.53%)|                                                        
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
                                                                                                                                                                                     
2018-04-12 13:08:24:651 INFO  [odb3] Publishing ONLINE status for database odb3.paysense...                                                                                          
2018-04-12 13:08:24:652 INFO  [odb3]->[[odb1]] Current copy of database 'paysense' is newer than the copy present in the cluster. Use the local copy and force other nodes to downloa
d this                                                                                                                                                                               
2018-04-12 13:08:24:654 INFO  [odb3] Received updated status odb1.paysense=NOT_AVAILABLE                                                                                             
2018-04-12 13:08:24:655 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                                                                                     
                                                                                                                                                                                     
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|Name   |Status|Databases                      |Conns|StartedOn|Binary           |HTTP         |UsedMemory              |                                                            
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|odb3(*)|ONLINE|paysense=ONLINE (MASTER)       |5    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|169.26MB/1.99GB (8.32%) |                                                        
|odb2(@)|ONLINE|paysense=NOT_AVAILABLE (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%)  |                                                        
|odb1   |ONLINE|paysense=ONLINE (MASTER)       |6    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|226.28MB/1.92GB (11.53%)|                                                        
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
                                                                                                                                                                                     
2018-04-12 13:08:24:655 INFO  [odb3] Recover complete for database 'paysense'                                                                                                        
2018-04-12 13:08:24:668 INFO  [odb3]<-[odb1] Received updated status odb2.paysense=SYNCHRONIZING                                                                                     

2018-04-12 13:08:24:668 INFO  [odb3]<-[odb1] Received updated status odb2.paysense=SYNCHRONIZING                                                                                     
2018-04-12 13:08:24:668 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                                                                                     
                                                                                                                                                                                     
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|Name   |Status|Databases                      |Conns|StartedOn|Binary           |HTTP         |UsedMemory              |                                                            
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|odb3(*)|ONLINE|paysense=ONLINE (MASTER)       |5    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|169.26MB/1.99GB (8.32%) |                                                        
|odb2(@)|ONLINE|paysense=NOT_AVAILABLE (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%)  |                                                        
|odb1   |ONLINE|paysense=NOT_AVAILABLE (MASTER)|6    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|226.28MB/1.92GB (11.53%)|                                                        
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
                                                                                                                                                                                     
2018-04-12 13:08:24:680 INFO  [odb3]->[odb2] Deploying database paysense...                                                                                                          
2018-04-12 13:08:24:681 INFO  [odb3]->[odb2] Creating backup of database 'paysense' (compressionRate=7) in directory: /tmp/orientdb/backup_paysense.zip...                           
2018-04-12 13:08:24:682 INFO  [odb3]->[odb2] Compressing database 'paysense' 29 clusters [internal, default, oschedule, device_1, device_2, e_1, device_3, e_3, e_2, knows, person_2,
 person_3, e, v_2, index, ouser, v_1, v_3, osequence, person_1, manindex, orole, v, person, knows_2, knows_1, ofunction, device, knows_3]...                                         
2018-04-12 13:08:24:683 INFO  [odb3] Received updated status odb3.paysense=BACKUP                                                                                                    
2018-04-12 13:08:24:684 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                                                                                     
                                                                                                                                                                                     
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|Name   |Status|Databases                      |Conns|StartedOn|Binary           |HTTP         |UsedMemory              |                                                            
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|odb3(*)|ONLINE|paysense=ONLINE (MASTER)       |5    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|169.26MB/1.99GB (8.32%) |                                                        
|odb2(@)|ONLINE|paysense=SYNCHRONIZING (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%)  |                                                        
|odb1   |ONLINE|paysense=NOT_AVAILABLE (MASTER)|6    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|226.28MB/1.92GB (11.53%)|                                                        
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
                                                                                                                                                                                     
2018-04-12 13:08:27:694 INFO  [odb3]->[odb2] - transferring chunk #1 offset=0 size=8388176 lsn={lastOperationTimeStamp:1523535132604,version:172,odb1:{segment:20,position:52},odb2:{
segment:27,position:52},odb3:{segment:29,position:52}}...                                                                                                                            
2018-04-12 13:08:27:694 INFO  [odb3]->[odb2] Deploy database task completed                                                                                                          
2018-04-12 13:08:30:220 INFO  [odb3]->[odb2] - transferring chunk #2 offset=8388176 size=432...                                                                                      
2018-04-12 13:08:30:240 INFO  [odb3]->[odb2] - transferring chunk #3 offset=8388608 size=8387947...                                                                                  
2018-04-12 13:08:31:737 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                                                                                     
                                                                                                                                                                                     
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|Name   |Status|Databases                      |Conns|StartedOn|Binary           |HTTP         |UsedMemory              |                                                            
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+                                                        
|odb3(*)|ONLINE|paysense=BACKUP (MASTER)       |5    |10:42:52 |10.10.15.141:2424|10.10.15.141:2480|169.26MB/1.99GB (8.32%) |                                                        
|odb2(@)|ONLINE|paysense=SYNCHRONIZING (MASTER)|2    |10:42:17 |10.10.15.156:2424|10.10.15.156:2480|1.09GB/2.00GB (54.41%)  |                                                        
|odb1   |ONLINE|paysense=NOT_AVAILABLE (MASTER)|6    |13:08:11 |10.10.15.179:2424|10.10.15.179:2480|226.28MB/1.92GB (11.53%)|                                                        
+-------+------+-------------------------------+-----+---------+-----------------+-----------------+------------------------+ 
2018-04-12 13:08:24:668 INFO  [odb3] Distributed servers status (*=current @=lockManager[odb2]):                                                                                                                                                                          
```
## Actual behavior  
Actually `odb1` should synchronize with either of `odb2` or `odb3`.   

## Steps to reproduce  
Remove one of the server (since all three are master and identical) and in our scenario we removed `odb1`.

Due to the above synchronization I am having downtime of few hours.

**Comment by tglman** [comment]
Hi @bipulkarnani,

We fixed a few synchronization issue with the 2.2.34, consider to upgrade if is possible.

Regards

**Comment by tglman** [comment]
Hi @bipulkarnani,

We fixed a few synchronization issue with the 2.2.34, consider to upgrade if is possible.

Regards
